article/app/services/dotcomponents/AMPPicker.scala

```diff
@@ -38,7 +38,21 @@ object AMPPageChecks extends Logging {
     }
   }
 
-  def isNotOpinion(page:PageWithStoryPackage): Boolean = ! page.item.tags.isComment
+  def isNotOpinion(page: PageWithStoryPackage): Boolean = ! page.item.tags.isComment
+
+  def hasOnlySupportedMainElements(page: PageWithStoryPackage): Boolean = {
+    def supported(block: BlockElement): Boolean = block match {
+      case _: ImageBlockElement => true
+      case _ => false
+    }
+
+    val hasMainVideo = for {
```

Comment:
In hindsight, this is poorly named.

Reply:
Not a showstopper to me at least, it'll get refactored in short order when we support more elements anyway!